test/RestrictedNORI-transfer.test.ts

```diff
@@ -0,0 +1,531 @@
+              restrictedAmounts[0] - amountToTransferFirstSchedule
+            ),
+            claimedAmount: BigNumber.from(0),
+            quantityRevoked: BigNumber.from(0),
+            exists: true,
+          },
+          {
+            tokenHolder: supplier,
+            scheduleTokenId: projectId2,
+            balance: BigNumber.from(
+              restrictedAmounts[1] - amountToTransferSecondSchedule
+            ),
+            claimableAmount: BigNumber.from(
+              restrictedAmounts[1] - amountToTransferSecondSchedule
+            ),
+            claimedAmount: BigNumber.from(0),
+            quantityRevoked: BigNumber.from(0),
+            exists: true,
+          },
+        ];
+        const expectedInvestor1ScheduleDetailsAfterTransfer = [
+          {
+            tokenHolder: investor1,
+            scheduleTokenId: projectId1,
+            balance: BigNumber.from(amountToTransferFirstSchedule),
+            claimableAmount: BigNumber.from(amountToTransferFirstSchedule),
+            claimedAmount: BigNumber.from(0),
+            quantityRevoked: BigNumber.from(0),
+            exists: true,
+          },
+          {
+            tokenHolder: investor1,
+            scheduleTokenId: projectId2,
+            balance: BigNumber.from(amountToTransferSecondSchedule),
+            claimableAmount: BigNumber.from(amountToTransferSecondSchedule),
+            claimedAmount: BigNumber.from(0),
+            quantityRevoked: BigNumber.from(0),
+            exists: true,
+          },
+        ];
+
+        for (const [
+          index,
+          scheduleDetail,
+        ] of supplierScheduleDetailsAfterTransfer.entries()) {
+          compareScheduleDetailForAddressStructs(
+            scheduleDetail,
+            expectedSupplierScheduleDetailsAfterTransfer[index]
+          );
+        }
+
+        for (const [
+          index,
+          scheduleDetail,
+        ] of investor1ScheduleDetailsAfterTransfer.entries()) {
+          compareScheduleDetailForAddressStructs(
+            scheduleDetail,
+            expectedInvestor1ScheduleDetailsAfterTransfer[index]
+          );
+        }
+        const [
+          supplierScheduleSetAfterTransfer,
+          investorScheduleSetAfterTransfer,
+        ] = await Promise.all([
+          rNori.getScheduleIdsForAccount(supplier),
+          rNori.getScheduleIdsForAccount(investor1),
+        ]);
+        expect(
+          supplierScheduleSetAfterTransfer.map((scheduleId) =>
+            scheduleId.toHexString()
+          )
+        ).to.have.deep.members([
+          BigNumber.from(projectId1).toHexString(),
+          BigNumber.from(projectId2).toHexString(),
+        ]);
+        expect(
+          investorScheduleSetAfterTransfer.map((scheduleId) =>
+            scheduleId.toHexString()
+          )
+        ).to.have.members([
+          BigNumber.from(projectId1).toHexString(),
+          BigNumber.from(projectId2).toHexString(),
+        ]);
```

Comment:
```suggestion
[supplierScheduleSetAfterTransfer, investorScheduleSetAfterTransfer].forEach((schedule) => { 
        expect(
          schedule.map((scheduleId) =>
            scheduleId.toHexString()
          )
        ).to.have.deep.members([
          BigNumber.from(projectId1).toHexString(),
          BigNumber.from(projectId2).toHexString(),
        ]);
})
```

Although might be an even better version of the above that doesnt require the forEach or schedule.map